windows/src/desktop/kmshell/main/OnlineUpdateCheck.pas

```diff
@@ -98,23 +102,141 @@   TOnlineUpdateCheck = class
 
     FShowErrors: Boolean;
     FDownload: TOnlineUpdateCheckDownloadParams;
-
+    {
+      Sets the download path and then calls the DoDownloadUpdates function. If
+      the user confirms the download (by clicking OK), the downloaded files are
+      marked to be deleted on the next system reboot. If the user cancels the
+      download, the downloaded files are deleted immediately.
```

Comment:
My memory was that we would download silently and in the background, rather than asking the user?

Reply:
This is adding comments to an existing function, it is foreground checking for updates. We will download updates in the background, however still allow checking for updates "now" by the user and choosing to download them. The UI object was tied to these downloads which made it hard to re-use the same functions.